spec/unit/network/http/api/indirected_routes_spec.rb

```diff
@@ -252,6 +252,8 @@
     end
 
     it "raises 406 not acceptable if no formats are accceptable" do
+      # do not want to stub out behavior in test
+      Puppet[:allow_pson_serialization] = true
```

Comment:
Since pson isn't available in main, the `handler` will try `json`, get an exception and then return 406 (as there are no more formats to try). 

To test that we don't fallback to pson by default, I'd suggest removing this line and instead change the test below to:

```ruby
    it "raises FormatError if tries to fallback" do
      data = Puppet::IndirectorTesting.new("my data")
```

To test that fallback to PSON can be enabled, I'd suggest updating the test above this one to:

```ruby
it "falls back to the next supported format", if: Puppet.features.pson? do`
   Puppet[:allow_pson_serialization] = true
   data = Puppet::IndirectorTesting.new("my data")
   ...
```

Since it's guarded by the `pson` feature, you'll need to create a `Gemfile.local`, add `gem 'puppet-pson'`  to it, run `bundle update` and then run the tests.

Reply:
Ah okay got it, by the test below, do you mean [this test](https://github.com/puppetlabs/puppet/blob/e98fa36a4991c77429ceb155afc8a8891f8f5800/spec/unit/network/http/api/indirected_routes_spec.rb#L254-L266) or [this test](https://github.com/puppetlabs/puppet/blob/e98fa36a4991c77429ceb155afc8a8891f8f5800/spec/unit/network/http/api/indirected_routes_spec.rb#L268-L280)?

Reply:
Ah sorry the latter one

Reply:
Ah nice, I'll change that test. For [this test](https://github.com/puppetlabs/puppet/blob/e98fa36a4991c77429ceb155afc8a8891f8f5800/spec/unit/network/http/api/indirected_routes_spec.rb#L254-L266), I needed to add `Puppet[:allow_pson_serialization] = true` because the test would fail without it. A FormatError would be raised instead of 406 although I'm not too sure why 🤔 

Here's how the test fails:

> expected Puppet::Network::HTTP::Error::HTTPNotAcceptableError with message matching /No supported formats are acceptable \(Accept: application\/json\)/, got #<Puppet::Network::FormatHandler::FormatError: Failed to serialize Puppet::IndirectorTesting for 'my'...son]: Could not render to Puppet::Network::Format[json]: source sequence is illegal/malformed utf-8> with backtrace:

Reply:
Ah, ok I see. I forgot that `Puppet[:allow_pson_serialization]=true` will allow the handler to continue trying all possible formats until the list is exhausted, thereby triggering the 406 response. I think it might be clearer to change that [test](https://github.com/puppetlabs/puppet/blob/main/spec/unit/network/http/api/indirected_routes_spec.rb#L254) to not be dependent on the `allow_pson_serialization` setting, something like:

```ruby
it "raises 406 not acceptable if no formats are accceptable" do
  data = Puppet::IndirectorTesting.new("my data")
  indirection.save(data, "my data")
  request = a_request_that_searches(Puppet::IndirectorTesting.new("my"), :accept_header => "unknown")
  
  expect {
    handler.call(request, response)
  }.to raise_error(Puppet::Network::HTTP::Error::HTTPNotAcceptableError,
                       %r{No supported formats are acceptable \(Accept: unknown)})
end
```

Reply:
Got it! I'll do that now, thank you for the help!